### DISCUSSION TITLE
Message Handler - Not Working with Filter.Command & Filter.Photo condition

### DISCUSSION THREAD

**Discussion by virajdesai3** [python-telegram-bot/python-telegram-bot#3328] [Q&A]
0

I am working on a image editing telegram bot - which can perform various functions based on commands like crop , removebg, etc.The idea is to accept single or multiple images from the user and get a command i.e. /crop or /removebg and get the bot to return edited images.

I am using the python-telegram-bot library and I am unable to trigger the message handler dispatcher defined as follows:
`dispatcher.add_handler(MessageHandler(Filters.command('crop') & (Filters.photo),crop))
`
whereas, something like Filters.caption works perfectly fine as below

`dispatcher.add_handler(MessageHandler(Filters.caption('crop') & (Filters.photo),crop))
`
However, it is practically impossible to add captions to each and every image when uploading multiple images.

Stackoverflow question [link](https://stackoverflow.com/questions/74276520/message-handler-function-with-filter-command-filter-photo-not-working)

**Comment by Poolitzer** [comment]
Sorry, the command is send after the images in a second message?

**Reply by virajdesai3** [reply]
Yes.

**Reply by Poolitzer** [reply]
A Message handler handles one update, not several. What you might want is a ConversationHandler setup, or some manual way to keep track of the file ids.

**Comment by virajdesai3** [comment]


**Reply by Bibo-Joshi** [reply]
Please avoid posting questions in multiple places at once. If you do, at least directly point that out in the question.

**Reply by virajdesai3** [reply]
thanks for pointing out. added links in both questions
